### DISCUSSION TITLE
What's your workflow when using NovelWriter?

### DISCUSSION THREAD

**Discussion by xahodo** [vkbo/novelWriter#1901] [General]
I'm curious. I'm finally learning to use NW to first actually outline my novel, before actually doing the writing also in NW.

Here's what I am doing at the moment:
1. Work out the main protagonist and various support characters.
2. Start each chapter and scene with a short `%synopsis`, filling out project notes where needed.
3. Add the various project notes I've collected to the scenes.
4. Take a look at the outline view. Does everything work out as nicely as I think it does. If not, fix the wonky parts.
5. Do the first draft.
6. Draft/edit cycle until satisfied (i.e. _never_ :D ).

What is your workflow with NW?

**Comment by vkbo** [comment]
I read about the Snowflake method many years ago, and found it useful as a very structured person by nature. I have my own adaptation of it though, so I don't really follow it closely. That's the idea of the method anyway. It's a proposal and a principle, not a strict model. (That is precisely why novelWriter doesn't have Snowflake forms like some other apps have added.)

Anyway ...

* I start with a plot idea, which I expand iteratively until I have a document (usually 10-15k words or so) detailing all the main plot points and turning points in the story. I like the classic three act model by the way, so it is also split up into acts.
* The full plot outline also has tags for each major plot point. So, I create the following plot tags in it: `IncitingEvent`, `KeyEvent`, `FirstPoint`, `FirstReaction`, `FirstPinch`, `SecondPoint`, `SecondAction`, `SecondPinch`, `ThirdPoint`, `PaceUp`, `Climax`, and finally `Resolution`.
* The plot also gives me the main characters, which I work on in between fleshing out the plot.
* Then I add supporting characters and additional world building. I also draw things, like maps and plans, and work out the necessary physics of things (I'm a physicist by training, so that's important to me).

After that, I've taken different approaches. I've tried to set up my key scenes in some earlier projects, based on the long form plot outline, but these days I just start writing scenes. Otherwise I just get bogged down into restructuring my scene outline over and over.

I sometimes write the key scenes (as defined by the plot tags above) a bit before I write the in-between scenes, but at this point in the process I find it better to leave the very structured approach behind and free-form a little. I treat the key scenes as my anchor points to make sure I stay on track. My process beyond this point isn't very well defined. I still need to figure out what works better.

Of course, I'm a hobby writer, not a professional writer, so my process isn't exactly finely tuned to being productive. I also have a lot of projects in various states.

**Reply by vkbo** [reply]
Well, with git you only push the changed files anyway, so it doesn't matter. Unless you mean if you should ignore some specific files?

The file structure of novelWriter is described here: https://novelwriter.io/docs/tech_storage.html

In short, your project structure is located in the `nwProject.nwx` file, so that is important. The text you've written is in the `content` folder. Nothing in the `meta` folder is important, maybe except for custom spell check words file and your session log if you like to keep track of your writing progress.

In general, if you restore data file in the `meta` folder to an older version that shouldn't be there, it will be processed and updated, or outright deleted if it's an index file (they can be regenerated). So in principle novelWriter should be able to handle file formats that have changed. All previous project file formats should also still be supported. There is a version number in it that tells the parser what format the data is in, and the different formats are regularly tested to ensure all older formats can be read.

So yeah, restoring individual files should also work, even if they are in an older format.

**Reply by vkbo** [reply]
Yeah, just make sure you always include the project structure file. But even if you don't, novelWriter will add content documents it doesn't know about to the project when it's opened. That's why there is some meta data saved at the top of the file.

The novelWriter project format is designed to be very robust and even if the app itself was broken or not available for whatever reason, you can still extract your text from the project folder with no losses. The files in the content folder are just plain text copies of what you've written, and the `ToC.txt` file contains a list of the order they were in. From those files, you can with a little effort (or a simple script), piece back together everything even without the app.

**Comment by HeyMyian** [comment]
I used to do outlining, first in Google Docs (bad) then in Scrivener (bad on a different level). It's also what I initially did when I made the switch to novelWriter. However, I found that writing down the outline killed my actual desire to work on the project. I suffer from thinking the job is as good as done once the outline is complete so I don't have to write it anymore. It's fatal. Sometimes, I still mis-use the comment feature within a document to give me very short notes on what future scenes I have planned. This comment is then carried over to the next document, and so on :sweat_smile:  Other than that, the outline is completely in my head.

As I'm writing, I'm still discovering new aspects, new subplots, new side characters, and literally go "oooh shiny" :smiley: It's what I love about this process, the discovery of things that weren't part of the initial idea. I think that's a lot harder to achieve (for me) with a written-down outline.

This means that I don't have a use of many novelWriter features. I also used to create character profiles and such, but after years of spending time with the characters I know them so well, profiles only hamper my creativity. All I use now is `@time` to track the year or date a scene takes place at (wrong use for this tag, I know), and `@char` to keep track of who is present in a scene in case I want an overview (never actually did want).

One project, which meant to be a short story of ca. 4k words, I wrote in a single document. It now has 20k words and is still a single document with scene headings; I prefer this setup in this case. Other projects that I know will be larger from the get-go have one document per chapter with optional scene headings within the document. But the documents are only created when I'm at the point of actually writing them. I write in chronological order and edit as I write. I'm not a fan personally of producing low-quality first drafts.

Ellen Brock has an (what I find) extremely insightful video about [The Four Types of Novel Writers](https://www.youtube.com/watch?v=eryQEZImm6Y) that gave me closure on my approach to writing.

**Reply by vkbo** [reply]
> As I'm writing, I'm still discovering new aspects, new subplots, new side characters, and literally go "oooh shiny" 😃 It's what I love about this process, the discovery of things that weren't part of the initial idea. I think that's a lot harder to achieve (for me) with a written-down outline.

I do that too. I only use the outline to make sure the story has a back bone, I. e. core plot driving moments. I'll admit I'm a bit of a traditionalist here in terms of what I expect from the flow of a plot.

I also don't use all keywords in novelWriter. I find `@char` and `@pov` very useful, and I track `@location` which I link to my extensive world building notes (I write sci-fi after all). I only occasionally use the other keywords.

As for your usage of `@time`, this is a better way of using it than the intended, which is mostly already covered by `@plot`, so I'm thinking of extending `@time` to support more time-like features.

**Comment by awqk** [comment]
I have this almost-finished novel, written using at least three different editors so far, VSCode being the last one. It's a set of text files (markdown), accompanied by a spreadsheet ordering the scenes, along with scene info: heading, PoV, single-line synopsis, word count, dependencies (must not precede/must come after, ...), status (1st draft, 2nd draft,...). Trying to keep the reader engaged is hard work, so there's another file tracking minor plot points (introduction, mention, closure, surprise-turn, follow-up), and a list of major plot points and which scenes advance them.

Since I found novelWriter, it's stalled, waiting for me to get around to migrate. Part of this is because I am not yet sure how to make the best use of novelWriter's tags and references, and how to track them. So I started writing short stories with novel writer to get a feel for things, and I am very happy with it.

Regarding workflow, I learned a thing or two:
- don't start without a sufficiently detailed outline.
- before changing your outline, think it through until the very end of your novel.
- don't start editing before writing the first draft from front to back.
- the beginning is the most important part of every story.

> Ellen Brock has an (what I find) extremely insightful video about The Four Types of Novel Writers that gave me closure on my approach to writing.

Very nice video. I see myself somewhere in the middle ground of the four extremes, with a tendency to "plotter". It may happen that a character grows on me, making me overturn things to please them.

**Comment by paulkater** [comment]
I see things called 'outline'. They don't exist in my world.

I'm a pantser. I get an idea for something and I start writing, usually with 1 or 2 characters somewhat in mind. As the characters grow, I add info to their personal character note, and facts and stuff go in a plot note. Only when I write a mystery (which sometimes happens), will I set up a spreadsheet with facts and moments in various story-lines, to keep track of who is where and what already happened. I can use that to work ahead a bit. Not too far though, because things always divert from what I plan.

[a comment by an automated account is not shown]

**Comment by Survivian** [comment]
My workflow is extremely erratic and subject to change as my circumstances change.

For one thing, I rarely - if ever - type while I have a dedicated program open in front of me to do it. I work a fairly boring office job and what I've ended up doing is stealing time away from them to literally anything else I'm interested in. Since the computer isn't mine, I can't really install software on it to do any writing and my screen is too exposed to do it with portable programs either, so I improvised. Everything I type first gets typed via Bluetooth keyboard into iOS Notes and then copy/pasted into my main editor of choice. I rarely have direct access to my notes.

For another, I used to use [QOwnNotes](https://github.com/pbek/QOwnNotes) for all my writing, having one file where the prose is and several others that contain notes to refer back to for it. It was also during this time that I would physically write all my prose and notes out in small notebooks and then manually transcribe them when I got home (same office job btw). This worked for a while, but when structure was starting to kick my ass, I looked for alternatives and pounced on this almost immediately.

I primarily write fanfiction, so I start with my specific flavors of the characters I'm yoinking and then make a very barebones outline of the largest plot.

Then I start making smaller plot documents for each of the main characters, I make plots that connect relevant plots between those characters, and I make world-building notes out of the logical questions that come out of those character connecting plots.

By the end, the most fleshed out parts of my notes are the world-building and the backstories. The plots are still mostly skeletons with beginnings, ends, and important events that happen in between; they're all set up chronologically and have references to all the other things to keep track of the relevant players.

Once all that's done, I start fleshing out the "macro" or main plot in chronological order, using references to all the smaller plots that happen inside it. And now I know exactly how the story will progress forwards, where the characters are, why they're doing it, and what they're doing. This leaves me room to fill details in my notes later because my best emergent ideas come out when I'm actually writing the thing, not planning it.

And since my fanfiction is released chapter by chapter, what I do is tag which of the smaller plots I want to focus on in the beginning of each chapter and chip away at the story piece by piece until I get to the end of the macro plot.

...Or at least that's the idea. I'm still in the process of setting up a future story like that and I'm currently dealing with import woes from QOwnNotes and applying retroactive outlines to a story that's getting some wicked continuity snarls from early, amateur decisions. And, mind you, this is still all being done while having to deal with some eventual weirdness that comes from "secretly" typing all this out on an iPhone first lmao.

It's a learning process, but I am getting better and that's all that matters.
